editor/src/core/model/element-metadata-utils.ts

```diff
@@ -503,8 +503,11 @@ export const MetadataUtils = {
     })
   },
   getRootViewPaths(elements: ElementInstanceMetadataMap, target: ElementPath): Array<ElementPath> {
-    const element = MetadataUtils.findElementByElementPath(elements, target)
-    return element?.rootElements ?? []
+    const allPathsUnordered = Object.keys(elements).map(EP.fromString)
+    const possibleRootElementsOfTarget = allPathsUnordered.filter((path) =>
+      EP.isRootElementOf(path, target),
+    )
```

Comment:
I fear that this is particularly expensive, especially on larger projects, and is now called very often. I'm not sure if this is something we should try to address via some additional caching, or by relying on the string representation (which honestly makes me feel very uneasy)